src/worker/indexers/chainIndexerRegistry.ts

```diff
@@ -1,67 +1,57 @@
 import cron from "node-cron";
-import { getConfig } from "../../utils/cache/getConfig";
-import { env } from "../../utils/env";
 import { getBlockTimeSeconds } from "../../utils/indexer/getBlockTime";
 import { logger } from "../../utils/logger";
-import { createChainIndexerTask } from "../tasks/chainIndexer";
+import { handleContractSubscriptions } from "../tasks/chainIndexer";
 
+// @TODO: Move all worker logic to Bullmq to better handle multiple hosts.
 export const INDEXER_REGISTRY = {} as Record<number, cron.ScheduledTask>;
 
 export const addChainIndexer = async (chainId: number) => {
   if (INDEXER_REGISTRY[chainId]) {
+    return;
+  }
+
+  // Estimate the block time in the last 100 blocks. Default to 2 second block times.
+  let blockTimeSeconds: number;
+  try {
+    blockTimeSeconds = await getBlockTimeSeconds(chainId, 100);
+  } catch (error) {
     logger({
       service: "worker",
-      level: "warn",
-      message: `Chain Indexer already exists: ${chainId}`,
+      level: "error",
+      message: `Could not estimate block time for chain ${chainId}`,
+      error,
     });
-    return;
+    blockTimeSeconds = 2;
   }
-
-  let processStarted = false;
-  const config = await getConfig();
-
-  // Estimate block time.
-  const blockTimeSeconds = await getBlockTimeSeconds(chainId);
-
-  const blocksIn5Seconds = Math.round((1 / blockTimeSeconds) * 5);
-  const maxBlocksToIndex = Math.max(
-    config.maxBlocksToIndex,
-    blocksIn5Seconds * 4,
-  );
```

Comment:
This was more complicated than it's worth. The goal was to find a good range of blocks to query. Honestly if we're realtime we're usually querying 5 seconds worth of blocks (2-20 depending on the block time). Hard coding this to a value greater than this but less than most eth_getLogs limits should suffice.